### PR TITLE
(chore) fix issues relative to sinequa libs modifications

### PR DESCRIPTION
**Pull request by hebus**
close https://sinequa.atlassian.net/browse/SBA-657

This PR fix the issues with the new sinequa libs

* sq-facet-tree is removed and replaced by sq-facet-list
* sq-facet-myseach is removed and replaced by sq-filters-view
* new integrated search-form component
* new autocomplete component

### CODE REVIEW

File: projects/vanilla/src/app/search/search.component.html [hunk 1]
```diff
@@ -153,8 +145,17 @@
...
+                    <sq-filters-view
+                      class="flex-grow-1 position-relative"
+                      [query]="searchService.query"
+                      [closeOnClickOutside]="true"
+                      [advancedModeClass]="'card position-absolute p-3'"
+                      (filterEdit)="searchService.search()">
+                    </sq-filters-view>
+                    <sq-results-counter [rowCount]="searchService.rowCount" class="mx-2"></sq-results-counter>
+                    <sq-sort-selector [results]="results" [rightAligned]="true"></sq-sort-selector>
+                </div>                </ng-template>
```
Comment (ericleib) [inline projects/vanilla/src/app/search/search.component.html]: I think this ng-template should contain only the sq-filters-view component, as the result-counter and sort-selector are in their own ng-template.

File: projects/vanilla/src/styles/preview.scss [hunk 2]
```diff
@@ -49,13 +49,13 @@ $highlights: map-merge(map-merge($extracts-highlights, $terms-highlights), $meta
 
 @mixin highlight($term) {
```
Comment (ericleib) [inline projects/vanilla/src/styles/preview.scss]: The changes in this file seem unrelated to the other. Did you commit them by mistake?

File: projects/vanilla/src/app/search/search.component.html [hunk 3]
```diff
@@ -14,15 +14,7 @@
...
-                <app-search-form
-                    [autocompleteSources]="config.autocompleteSources"
-                    [enableVoiceRecognition]="config.enableVoiceRecognition"
-                    [keepTab]="config.keepTab"
-                    [enableKeepFilters]="config.enableKeepFilters"
-                    [keepFilters]="config.keepFilters"
-                    [enableAdvancedForm]="config.enableAdvancedForm"
-                    [keepAdvancedSearchFilters]="config.keepAdvancedSearchFilters">
-                </app-search-form>
+                <app-search-form [autocompleteSources]="config.autocompleteSources" [searchRoute]="config.searchRoute"></app-search-form>
```
Comment (ericleib) [inline projects/vanilla/src/app/search/search.component.html]: With all these options being removed, we should remove them from the preview configurator as well.

File: projects/vanilla/src/app/app.module.ts [hunk 4]
```diff
@@ -77,10 +77,14 @@ import { ConfiguratorsModule } from "./configurators/configurators.module";
...
 
+// standalone components
+import { SearchFormComponent } from "@sinequa/components/search-form";
+import { FiltersModule } from "@sinequa/components/filters";
+
 
 // Initialization of @sinequa/core
 export const startConfig: StartConfig = {
-    //app: "training",
+    app: "training",
```
Comment (ericleib) [inline projects/vanilla/src/app/app.module.ts]: This comment should not be removed, or else the app won't work when packaged in the product.

File: projects/vanilla/src/app/app.module.ts [hunk 5]
```diff
@@ -77,10 +77,14 @@ import { ConfiguratorsModule } from "./configurators/configurators.module";
 import { ConditionsModule } from "@sinequa/ngx-ui-builder";
 import { AppConfigService } from "./app-config.service";
 
+// standalone components
+import { SearchFormComponent } from "@sinequa/components/search-form";
+import { FiltersModule } from "@sinequa/components/filters";
```
Comment (ericleib) [inline projects/vanilla/src/app/app.module.ts]: (the filters module is not a standalone component, just a normal module)

File: projects/vanilla/src/app/search/search.component.html (hunk 3 above)
Comment (ericleib) [reply projects/vanilla/src/app/search/search.component.html]: Also, "searchRoute" should not be part of the config, since we don't want the user to customize this.

**Review by ericleib — CHANGES_REQUESTED** [review]

File: projects/vanilla/src/styles/preview.scss (hunk 2 above)
Comment (hebus) [reply projects/vanilla/src/styles/preview.scss]: You right... I will remove it

File: projects/vanilla/src/app/search/search.component.html [hunk 6]
```diff
@@ -152,8 +144,16 @@
                     <sq-tabs [results]="results" [showCounts]="!config.hideCounts"></sq-tabs>
                 </ng-template>
 
-                <ng-template uib-template="my-search" display="My search" let-config>
-                    <sq-facet-mysearch [results]="results" [allowDeletion]="config.allowDeletion"></sq-facet-mysearch>
+                <ng-template uib-template="filters-view" display="Filters view" let-config>
+                  <div class="d-flex small sq-toolbar">
```
Comment (ericleib) [inline projects/vanilla/src/app/search/search.component.html]: I think that you can remove this div (the "small" class could go in the container configuration)

**Review by ericleib — CHANGES_REQUESTED** [review]

File: projects/vanilla/src/app/search/search.component.html (hunk 6 above)
Comment (hebus) [reply projects/vanilla/src/app/search/search.component.html]: done

**Review by ericleib — APPROVED** [review]
